### ISSUE TITLE
Input_chunk_length

### ISSUE THREAD

**Issue by chelvanai** [unit8co/darts#848]
Hi
Why darts uses input_chunk_length for prediction; didn’t  it take the last point of training_length to prediction? If input_chunk_length is less than training_length some points will ignore in prediction. 

Is my question is right?
Please give some explain.

**Comment by chelvanai** [comment]
import numpy as np
from darts.utils.timeseries_generation import datetime_attribute_timeseries

from darts.dataprocessing.transformers import Scaler
from darts.metrics import mape
from darts.timeseries import TimeSeries
from darts.models import RNNModel
import matplotlib.pyplot as plt
import pandas as pd

series = np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25])

series = TimeSeries.from_values(series)

train, val = series.split_after(20)

transformer = Scaler()
series_transformed = transformer.fit_transform(series)

my_model = RNNModel(
    model="LSTM",
    hidden_dim=20,
    dropout=0,
    batch_size=2,
    n_epochs=300,
    optimizer_kwargs={"lr": 1e-3},
    log_tensorboard=True,
    random_state=42,
    training_length=4,
    input_chunk_length=4,
    force_reset=True,
    save_checkpoints=True,
)

my_model.fit(
    series_transformed,
    verbose=True,
)

pred = my_model.predict(n=5)

print(transformer.inverse_transform(pred))


how to know the training shift data in this code?

**Comment by mrasidbakir** [comment]
I think you are asking how to determine the look back value. According to darts designers, this value should be determined by periodicity.

**Comment by chelvanai** [comment]
I am analyzing the utils; data shift code because normal RNN we will use sliding window for shift data, but they use it in different way.
